### PR TITLE
Support ghc 9.8

### PR DESCRIPTION
**Pull request by alexfmpe**
On top of #31 but done as a separate PR as the `dependent-sum-template` lower bound is increased, which I guess can count as a breaking change?

EDIT: Oh probably not, since it's only a bound on the executable?
I tried only increasing the upper bound but it keeps picking up 0.1 unless I increase the lower one.
For the record, here's the error
```
Building library for dependent-sum-template-0.1.1.1..
[1 of 3] Compiling Data.Dependent.Sum.TH.Internal ( src/Data/Dependent/Sum/TH/Internal.hs, dist/build/Data/Dependent/Sum/TH/Internal.o, dist/build/Data/Dependent/Sum/TH/Internal.dyn_o )

src/Data/Dependent/Sum/TH/Internal.hs:40:22: error: [GHC-83865]
    • Couldn't match type ‘BndrVis’ with ‘()’
      Expected: [TyVarBndrUnit]
        Actual: [TyVarBndr BndrVis]
    • In the first argument of ‘f’, namely ‘bndrs’
      In a stmt of a 'do' block: dec <- f bndrs cons
      In the expression:
        do dec <- f bndrs cons
           return [InstanceD overlaps cxt classHead [...]]
   |
40 |             dec <- f bndrs cons
   |                      ^^^^^

src/Data/Dependent/Sum/TH/Internal.hs:46:17: error: [GHC-83865]
    • Couldn't match type ‘BndrVis’ with ‘()’
      Expected: [TyVarBndrUnit]
        Actual: [TyVarBndr BndrVis]
    • In the first argument of ‘f’, namely ‘bndrs’
      In the expression: f bndrs cons
      In an equation for ‘dec’: dec = f bndrs cons
   |
46 |         dec = f bndrs cons
   |                 ^^^^^
Error: cabal: Failed to build dependent-sum-template-0.1.1.1 (which is
required by exe:readme from aeson-gadt-th-0.2.5.1). See the build log above
for details.
```

### CODE REVIEW

**Comment by alexfmpe** [comment]
Think this also needs a release of 
https://github.com/obsidiansystems/dependent-sum-template/commit/dfaf678c10d1032d5870658c8777d58e1c965202

**Review by ali-abrar — APPROVED** [review]

File: release.nix [hunk 1]
```diff
@@ -1,103 +1,21 @@
...
-          sha256 = "0rm91akq3d4b8xa127dklgd1vg2x9xv962pg98i7xhgnllp6i5r3";
-        } {});
-        quickcheck-instances = self.callHackageDirect {
-          pkg = "quickcheck-instances";
-          ver = "0.3.23";
-          sha256 = "1baqh2harkcx7igqmk6p040vmchy30wnh1crwwvzcxqv22iwyfrw";
+let pkgs = import ./nixpkgs/23.11 {};
+    hpkgs = {
+      overrides = self: super: {
+        dependent-sum-template = self.callHackageDirect {
```
Comment (alexfmpe) [inline release.nix:4]: @ali-abrar  Might be able to just do
https://github.com/NixOS/nixpkgs/blob/e22b1a7d7b0e338cca6932ab5a3f16e20fcf1c32/pkgs/development/haskell-modules/configuration-ghc-9.8.x.nix#L91

Comment (ali-abrar) [reply release.nix:4]: yeah, probably. but i already merged ;)

Comment (alexfmpe) [reply release.nix:4]: RIght, I meant it more as a FYI for any other repos with a big blob of release.nix that can now be aggressively pruned down

Comment (ali-abrar) [reply release.nix:4]: thanks for all the PRs btw

**Comment by ali-abrar** [comment]
released to [hackage](https://hackage.haskell.org/package/aeson-gadt-th-0.2.5.2)

**Pull request merged**